### ISSUE TITLE
Request new MD: Ordered list indentation (ol-indent)

### ISSUE THREAD

**Issue by GandalfSaxe** [DavidAnson/markdownlint#138]
From the discussion in #137 we seemed to agree that a rule for ordered list indentation (alias: `ol-indent`) is oddly missing. Rule [MD007](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md007) sets a default spaces to 2 for ordered sublists, however there is no such constraint on unordered sublists.

So while current we would only get a warning to the first (unordered) list:
```markdown
* one
* two
* three
    * sub1
    * sub2

1. one
2. two
3. three
    1. sub1
    2. sub2
```

this issue suggests that `ol-indent` would also gives warning for the second (ordered) list for using 4 spaces, whereas the correct would be 3.

**Comment by TheJaredWilcurt** [comment]
Would this also handle this case?

```md
1. Text

    ![This should be flagged for starting with 4 spaces instead of 2](file.png)

1. Test
```

**Comment by jthomerson** [comment]
Any update on this rule? If a PR were created that mimicked the behavior of ul-indent, but worked with ol's, would that be reviewed and accepted?

**Comment by DavidAnson** [comment]
I have no objections to this proposed rule, but have not gotten around to adding it myself. An implementation that is structured like MD007 but applies to ordered lists seems like it would be suitable here. I’d be happy to review such a PR according to the [contribution guidelines](https://github.com/DavidAnson/markdownlint/blob/master/CONTRIBUTING.md).

**Comment by mschoettle** [comment]
Would it make sense to have a generic indentation rule?

For example:

````markdown
* A ul item

  ```python
  print('Hello World')
  ```
````

Does not render the code block indented with the ul item. Maybe it depends on the renderer though.
In any case, it could be helpful to enforce general consistent indentation.

**Comment by DavidAnson** [comment]
FYI, the example immediately above should/does render the code fence as part of the list item (per CommonMark spec): https://dlaa.me/markdownlint/#%25m%23%20Example%0A%0A*%20A%20ul%20item%0A%0A%20%20%60%60%60python%0A%20%20print('Hello%20World')%0A%20%20%60%60%60%0A%0A%20%20Still%20item%0A%0ANew%20paragraph%0A%0A%60%60%60python%0Aprint('Hello%20World')%0A%60%60%60%0A

**Comment by DavidAnson** [comment]
Okay. FYI that I recently had a somewhat related discussion and the second half of the messages in this issue is relevant: https://github.com/DavidAnson/markdownlint/issues/1473

Specifically, I didn't see an easy way to get list item indent from the parser. However in your case, I wouldn't even be able to trust the parser because you want to pick up on cases where TOO FEW spaces are used and the list ends early. That would push the implementation over toward regular expressions which is both a blessing and a curse. :)
